File: java/Runner.java
import java.io.*;
import java.util.*;

public class Runner {
  public static void main(String[] args) {
    BufferedReader bw = new BufferedReader(
                          new InputStreamReader(System.in));
    Heap<Integer> heap = new Heap<Integer>();

    String line;
    try {
      while((line = bw.readLine()) != null) {
        String[] tokens = line.split(" ");
        int mode = Integer.parseInt(tokens[0]);

        switch(mode) {
          case 0:
            int value = Integer.parseInt(tokens[1]);
            heap.insert(value);
            break;
          case 1:
            System.out.println(heap.top());
            break;
          case 2:
            heap.pop();
            break;
        }
      }
    } catch (IOException ex) {
      System.err.println("Exception: " + ex.getLocalizedMessage());
    }
  }
}

File: csharp/Runner.cs
using System;

public class Runner {
  public static void Main(String[] args) {
    Heap<int> heap = new Heap<int>();
    
    String line;
    while((line = Console.ReadLine()) != null) {
      String[] tokens = line.Split();
      int mode = Int32.Parse(tokens[0]);

      switch(mode) {
        case 0:
          int val = Int32.Parse(tokens[1]);
          heap.Insert(val);
          break;
        case 1:
          Console.WriteLine(heap.Top());
          break;
        case 2:
          heap.Pop();
          break;
        }
    }
  }
}
